Add boston housing example

davidjurado opened the pull request:
This example is a tutorial on how to pack an existing project into MLCube using config 2.0

Please refer to the [README](https://github.com/davidjurado/mlcube_examples/blob/feature/boston_housing/boston_housing/README.md).

davidjurado committed:
Add boston housing example

davidjurado committed:
Update README.md

bitfort commented on boston_housing/mlcube/.mlcube.yaml:
```diff
@@ -0,0 +1,29 @@
+name: MLCommons Boston Housing
```
I don't follow why this is called ".mlcube.yaml"; whats the motivation for prefixed "."?

davidjurado replied:
The ".mlcube.yaml" file corresponds to the packed version for the MLCube definition, this file contains the needed information to unpack the full MLCube original structure for folders and files using the command: `mlcube describe`. This will also generate a file called "mlcube.yaml" (without the initial dot) that is the original MLCube structure definition file.

I'm currently working on the implementation of config 2.0 for this tutorial, when this is done we won't need to extract the MLCube original folder structure, and instead, there will be only a single file called: "mlcube.yaml" with all the needed information.

bitfort commented on boston_housing/README.md:
```diff
@@ -0,0 +1,216 @@
+# Packing an existing project into MLCUbe
```
Great job going through explaining things in detail!

bitfort commented on boston_housing/project/01_download_dataset.py:
```diff
@@ -0,0 +1,34 @@
+"""Download the raw Boston Housing Dataset"""
```
Love the simple straight forward files you built here!

bitfort commented on boston_housing/project/mlcube.py:
```diff
@@ -0,0 +1,73 @@
+"""MLCube handler file"""
```
This is a great main file - and I'm thinking about how we can further simplify this in the spec; nothing need to change here, but long term we should think about how to make this file simple for people to create :)

davidjurado committed:
Update to config 2.0

davidjurado committed:
Add support for overriding parameters at command line

davidjurado committed:
Update config file to v2.0

davidjurado committed:
Fix dockerfile

davidjurado committed:
Update Readme

davidjurado committed:
Fix Readme

sergey-serebryakov committed:
Bug fix: MLCube example never recognized it was running for the 1st time. (#35)

The greeting message printed by this MLCube can differ if it runs for 1st time or not. It identifies this by checking if a chat file exists. The MLCube was implemented so that it was checking for file existence after opening this file, so the function that checks if file exists or not was always returning true.

davidjurado committed:
Merge branch 'feature/boston_housing' of https://github.com/davidjurado/mlcube_examples into feature/boston_housing

davidjurado committed:
Update MLCube installation command in Readme
